### PR TITLE
Modified feature store managed VNet sample notebook.

### PR DESCRIPTION
**Pull request by ynpandey**
# Description

# Checklist

- [ ] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb [hunk 1]
```diff
@@ -320,20 +346,19 @@
    },
    "source": [
     "## Note\n",
-    "Feature store Vs Project workspace: You will use a featurestore to reuse features across projects. You will use a project workspace(the current workspace) to train and inference models, by leveraging features from feature stores. Many project workspaces can share and reuse a same feature store.\n",
+    "__Feature store vs Project workspace__: You will use a feature store to reuse features across projects. You will use a project workspace (the current workspace) to train and inference models, by leveraging features from the feature stores. Many project workspaces can share and reuse the same feature store.\n",
```
Comment (qjxu) [inline sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:349]: You will use a feature store workspace ...

"workspace" is missing.

File: sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb [hunk 2]
```diff
@@ -515,17 +579,26 @@
     "#### Step 1b. Copy the sample data required for this tutorial series to the newly created storage containers."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To copy sample data to the source data storage container, please assign **Storage Blob Data Contributor** and **Contributor** roles to the user identity. The following cell copies sample source data for transactions feature set used in this tutorial from a public storage account to the newly created storage account."
```
Comment (qjxu) [inline sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:586]: please assign *Storage Blob Data Contributor** and **Contributor** roles...

Duplicate?

Comment (qjxu) [reply sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:586]: Do you have the cli command for the role assignment?

File: sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb [hunk 3]
```diff
@@ -889,17 +1038,19 @@
    },
    "outputs": [],
    "source": [
-    "!az role assignment create --role \"Storage Blob Data Reader\" --assignee-object-id $uai_principal_id --assignee-principal-type ServicePrincipal --scope $source_data_gen2_container_arm_id"
+    "!az role assignment create --role \"Storage Blob Data Contributor\" --assignee-object-id $uai_principal_id --assignee-principal-type ServicePrincipal --scope $source_data_gen2_container_arm_id"
```
Comment (qjxu) [inline sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:1041]: role assignment should be done before the copy.

File: sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb (hunk 2 above)
Comment (ynpandey) [reply sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:586]: No, both Contributor and Storage Blob Data Contributor roles are needed. For the reasons listed here https://learn.microsoft.com/en-us/azure/storage/blobs/assign-azure-role-data-access?tabs=portal#assign-an-azure-role, Reader or Contributor is needed beside Storage Blob Data Contributor role.

File: sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb (hunk 3 above)
Comment (ynpandey) [reply sdk/python/featurestore_sample/notebooks/sdk_and_cli/network_isolation/Network Isolation for Feature store.ipynb:1041]: This is mentioned in the Step 1b text before code cells. Removed the role assignment in the later part as it is redundant.

**Review by shail2208 — APPROVED** [review]

**Review by qjxu — APPROVED** [review]

**Pull request merged**